### PR TITLE
Ks 93 Create MessageCard component

### PR DESCRIPTION
**Pull request by Tommoore96**
# Description
Creation of the MessageCard component.  Ticket can be found [here](https://k4o.atlassian.net/browse/KS-93).

Component looks like and is used in this manner: 
image

Matches designs outlined [here](https://www.figma.com/file/KxwTJZOx3qhe74R14fzhXg/Koodoo-Components?node-id=1034%3A318).

## To run
1. Install bit with these [instructions](https://harmony-docs.bit.dev/getting-started/installing-bit/#install-bvm)
2. `bit import`
3. `bit install`
4. `bit start`
5. Head to [http://localhost:3000/ui/message-card](http://localhost:3000/ui/message-card)

## Philosophical Musings
You'll notice in `koomeleon/ui/message-card/message-card.tsx` that I'm being very lean and going almost logicless, leveraging css & styled components as much as possible.  Instead of a component that takes props and renders things conditionally, it is a single wrapper (just as a div really is) that decorates html code put inside it.  There is no logic except that which checks the children for an SVG.  I've managed to use TS to insist on the type of html passed in, 1 optional `svg` element and 2 `p` elements.  I think this is perfect for this component as it's such a simple component.

Perhaps in more complex components we won't be able to ask them to use raw HTML as children, maybe they'll want differently styled children of the same element type (eg `ul`) styled differently; but that doesn't mean we should revert back to providing a single component that takes a load of props.  We can simply provide them with more than one wrapper element from the component service like so:
```js
<Menu>
  <Menu.Item>Home</Menu.Item>
  <Menu.Section>Products</Menu.Section>
</Menu>
```
There is also the option of providing classNames for people to use on the passed in html/jsx code, as the styled components can easily access the classNames of their children.  Both approaches would prevent us from having to accept arrays of objects and mapping over them and turning them into components (which is how Koomeleon 1 would handle it).  It also gives control to the monolith app over which children to display and when.

However we decide to do it, I think the days of providing blank components that accept a load of props, do a load of calcs and spit out magic should be over.  Let the monolith app handle the components and let the components handle what they look like in in their situations.  If there's any complex logic that's constantly used across apps then that's for a utils library and not a components library.

### CODE REVIEW

File: koomeleon/ui/message-card/message-card.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+export const MessageCard = styled.div<MessageCardProps>`
+  background: #ffffff;
+  border-radius: 6px;
+  box-shadow: 0px 0px 0px 1px #e2e5ed;
+  color: #1f1f1f;
+  padding: 16px;
+
+  ${({ children }) =>
+    children[0].type === 'svg'
```
Comment (stevelouiscooper) [inline koomeleon/ui/message-card/message-card.tsx:35]: Will this work if nobody passes any children? I always get worried when I see array indexes referenced directly in code.

**Review by stevelouiscooper — APPROVED** [review]
Looking good, and nice comments. I think your implementation is clean other than what I mentioned below, and the philosophical approach makes sense to me though it seems like we're going from one extreme to the other (too many props to no props at all) and I'm not sure that's entirely necessary. Props can be useful - it can be nice to know what you're supposed to pass to a component and getting feedback when you're doing it wrong. I don't think you're getting as much with a pure CSS implementation (no intellisense, etc). I think if we are to go with this approach to implementation I'll have to refactor a bunch of components and I'm not sure how long that'll take as I've never built components this way but it definitely feels like a solid approach. Good stuff 👍

**Comment by aranbruce** [comment]
@Tommoore96 Great write up! Think this approach is really interesting and it's great to see the flexibility this can give us. 

TL;DR to my challenges - just because we can make our components flexible should we?

Playing devil's advocate to this approach for a second, a couple of questions from me would be:

- Is there a danger that this added flexibility means that we risk becoming unaligned between design and code e.g. if our code can do a bunch of things our designs can't are we making it hard for the 2 to mean the same thing when we refer to them. An example here is that hasn't this component just become a generic card component now?
- Do we actually want to enforce some design decisions in our components so that they don't quickly become bastardised? We've been quite thoughtful in the designs with regard to how we've constructed our components a certain way. For example in this component, I wouldn't want someone adding more than one title and one paragraph as that would look a bit 💩. Think we need to think about if we **should** facilitate use-cases that have not been "designed"
- If we get to a use case where we need a component to be more flexible can't we update it then?
- Agree with Steve that having a bit more "opinionated" components can also make it a devs life easier and make it harder to go wrong with them as per the example I gave above.
- Is there a danger that making our components too flexible does away with much of the advantage of having components at all? Extreme (and slightly silly) example but why not just have divs?

Great to see this post as it's an important conversation to have and am just trying to think critically about the choices we make.

Happy to chat about any of the above through
